test/unit/file_transfer_agent/azure_test.js

```diff
@@ -0,0 +1,363 @@
+  const mockKey = 'mockKey';
+  const mockIv = 'mockIv';
+  const mockMatDesc = 'mockMatDesc';
+
+  let AZ = null;
+  let azure = null;
+  let filestream = null;
+  const dataFile = mockDataFile;
+  const meta = {
+    stageInfo: {
+      location: mockLocation,
+      path: mockTable + '/' + mockPath + '/',
+      creds: {}
+    },
+    SHA256_DIGEST: mockDigest,
+  };
+  const encryptionMetadata = {
+    key: mockKey,
+    iv: mockIv,
+    matDesc: mockMatDesc
+  };
+
+  before(function () {
+    mock('azure', {
+      BlobServiceClient: function() {
+        function BlobServiceClient() {
+          this.getContainerClient = function () {
+            function getContainerClient() {
+              this.getBlobClient = function () {
+                function getBlobClient() {
+                  this.getProperties = function () {
+                    function getProperties() { 
+                      this.then = function (callback) {
+                        callback({
+                          metadata: {}
+                        });
+                      };
+                    }
+                    return new getProperties;
+                  };
+                }
+                return new getBlobClient;
+              };
+            }
+            return new getContainerClient;
+          };
+        }
+        return new BlobServiceClient;
+      }
+    });
+
+    mock('filestream', {
+      readFileSync: async function (data) {
+        return data;
+      }
+    });
+    
+    azure = require('azure');
+    filestream = require('filestream');
+    AZ = new SnowflakeAzureUtil(azure, filestream);
+  });
+
+  it('extract bucket name and path', async function () {
+    let result = AZ.extractContainerNameAndPath('sfc-dev1-regression/test_sub_dir/');
+    assert.strictEqual(result.containerName, 'sfc-dev1-regression');
+    assert.strictEqual(result.path, 'test_sub_dir/');
+
+    result = AZ.extractContainerNameAndPath('sfc-dev1-regression/stakeda/test_stg/test_sub_dir/');
+    assert.strictEqual(result.containerName, 'sfc-dev1-regression');
+    assert.strictEqual(result.path, 'stakeda/test_stg/test_sub_dir/');
+
+    result = AZ.extractContainerNameAndPath('sfc-dev1-regression/');
+    assert.strictEqual(result.containerName, 'sfc-dev1-regression');
+    assert.strictEqual(result.path, '');
+
+    result = AZ.extractContainerNameAndPath('sfc-dev1-regression//');
+    assert.strictEqual(result.containerName, 'sfc-dev1-regression');
+    assert.strictEqual(result.path, '/');
+
+    result = AZ.extractContainerNameAndPath('sfc-dev1-regression///');
+    assert.strictEqual(result.containerName, 'sfc-dev1-regression');
+    assert.strictEqual(result.path, '//');
+  });
+
+  it('get file header - success', async function () {
+    await AZ.getFileHeader(meta, dataFile);
+    assert.strictEqual(meta['resultStatus'], resultStatus.UPLOADED);
+  });
+
+  it('get file header - fail expired token', async function () {
+    mock('azure', {
```

Comment:
in our tests we mock finally `getProperties.then` function - can we extract a helper method and only pass this function to it?

Reply:
updated